### DISCUSSION TITLE
Xiaomi 12 can not see network (5G SA)

### DISCUSSION THREAD

**Discussion by jarroyogrd** [srsran/srsRAN_Project#189] [General Help] [answered]
Hello community,

I am working on getting a Xiaomi 12 smartphone working with srsrRAN_Project. 

### Setup
- USRP B210 with GPSDO
- Xiaomi 12
- Sysmocom SJA2 programmable SIM cards
- Open5GS core

### Description
I have other UEs such as the Quectel RM500Q working correctly (they are registered in the network and send/receive traffic as expected), however the Xiaomi 12 is not able to connect to the network or even see it.

I have followed the documentation for COTS UEs (https://docs.srsran.com/projects/project/en/latest/tutorials/source/cotsUE/source/index.html ) with no success. I have also tried the following:
- Different PLMNs. I have tried 99970 (with that PLMN in my SIM), 90170 and 90101 (in order to force roaming as explained in the documentation, with PLMNs 99970 and 00101 configured in the SIM for each case), and 00101 (as described in the list of tested devices for the Xiaomi 12, with the matching PLMN programmed in the SIM).
- Adjust the gains of the gNB. I have tried many combinations of TX/RX gains for the gNB.
- Inserting the SIM in both slots.

I put the following configurations in my Xiaomi 12:
- Prefer 5G connections
- Enable data roaming
- Disable VoLTE and VoNR
- APN configuration 
- Set SA option

The list of tested devices includes the Xiaomi 12. Does anyone know if the phone needs any additional configuration? 

I attach the AMF, NSSF and gNB configuration files, along with the Xiaomi 12 configurations screenshots in configuration.zip. I also attach relevant logs and traces of our working scenario (with a Quectel RM500Q) in quectel.zip and our problematic scenario (Xiaomi 12) in xiaomi.zip.

Thanks in advance.

Best regards,

Jorge

Note 1: In the logs and traces I did not use a GPSDO with the USRP since it was not available when I generated them, but from the gNB side I understand that everything works correctly since the Quectel RM500Q can register to the network as expected without a GPSDO. Anyways, I saw the same behavior with and without a GPSDO.

Note 2: I had to remove some parts of the Quectel logs and traces since the files are too big to upload to GitHub.

### Files
[configuration.zip](https://github.com/srsran/srsRAN_Project/files/12463008/configuration.zip)
[quectel.zip](https://github.com/srsran/srsRAN_Project/files/12463011/quectel.zip)
[xiaomi.zip](https://github.com/srsran/srsRAN_Project/files/12463015/xiaomi.zip)

**Comment by pgawlowicz** [comment]
Hi @jarroyogrd any update on this issue?

**Reply by jarroyogrd** [reply]
Hello @pgawlowicz , sorry for the late reply. We were not able to connect the Xiaomi 12 to srsRAN.

**Reply by pgawlowicz** [reply]
thanks for your feedback. btw. did you manage to connect any COTS phone to your srsRAN network?

**Reply by jarroyogrd** [reply, marked as answer]
Yes, we ended up purchasing the OnePlus 8T in order to follow the COTS UE tutorial in the srsRAN Project documentation and it worked as expected. We have also successfully connected a Quectel RM500Q module as I mentioned in the discussion.
